rust/routee-compass-core/src/model/unit/energy_unit.rs

```diff
@@ -84,13 +84,26 @@ impl FromStr for EnergyUnit {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         use EnergyUnit as E;
-        match s.trim().to_lowercase().replace("_", " ").as_str() {
-            "gallons gasoline" | "gas" => Ok(E::GallonsGasoline),
-            "gallons diesel" | "diesel" => Ok(E::GallonsDiesel),
-            "kilowathours" | "kilowatthour" | "kwh" => Ok(E::KilowattHours),
-            "liters gasoline" => Ok(E::LitersGasoline),
-            "liters diesel" => Ok(E::LitersDiesel),
+        match s
+            .trim()
+            .to_lowercase()
+            .replace("_", " ")
+            .replace(" ", "")
+            .as_str()
+        {
+            "gallonsgasoline" | "gas" => Ok(E::GallonsGasoline),
```

Comment:
I'm wondering if it might be too ambiguous to allow "gas" to translate to gallons but not to liters? Perhaps it's better to require that you specify gallons or liters so it's not unclear about which one it maps to? Same with diesel.

Reply:
you're right that this is definitely a US-centric take on a common abbreviation. i'll do as you suggest and remove "gas" and "diesel" aliases.